### PR TITLE
Add 65k GKE benchmark

### PR DESCRIPTION
**Pull request by besher-massri**
This PR adds a benchmark for GKE on 65,000 nodes scale

This is following up on the recent [GKE announcement](https://cloud.google.com/blog/products/containers-kubernetes/gke-65k-nodes-and-counting) for support clusters with 65,000-nodes scale.

It benchmarks GKE on a (simulated) AI workload using using Terraform and ClusterLoader2.

I didn't include it as part of `benchmarks` directory as the benchmarking process here is different (using CL2) and has a different infra setup.

### CODE REVIEW

**Comment by andrewsykim** [comment]
Agree with @achandrasekar that it would be good to put this under the existing `benchmarks/` directory. Is it possible to make that change?

File: 65k-benchmarks/simulated-ai-workload/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  --v=2
+```
+
+The flag `--enable-prometheus-server=true` deploys prometheus server using prometheus-operator.
+
+Make sure that `--testconfig` flag points to the correct path of the benchmark.
+
+## Results
+
+The benchmark results are stored in the `./output/` directory of the `perf-tests` repository. You can use these results to analyze the performance and scalability of your GKE cluster.
```
Comment (andrewsykim) [inline 65k-benchmarks/simulated-ai-workload/README.md:118]: Do we plan to publish example results from one of the runs?

**Comment by besher-massri** [comment]
> Agree with Ashok Chandrasekar that it would be good to put this under the existing `benchmarks/` directory. Is it possible to make that change?
> 
> > I didn't include it as part of benchmarks directory as the benchmarking process here is different (using CL2) and has a different infra setup.
> 
> I think tooling being different is not a blocker, but would be good to consoildiate eventually

I had an attempt at reorganizing the benchmark, here is a summary of the changes:
- put existing infra under infra/gpu-cluster
- moved infra part under infra, put under folder 65k-cpu-cluster
- moved CL2 part under benchmark/tools
- renamed current README to gpu-benchmark.md
- moved 65k README and renamed into 65k-nodes-simulated-ai-workload
- added a root level README for `benchmarks/` directory

File: 65k-benchmarks/simulated-ai-workload/README.md (hunk 1 above)
Comment (besher-massri) [reply 65k-benchmarks/simulated-ai-workload/README.md]: Not here, but part of the blogpost. The idea is to link to the blogpost once published.

**Comment by achandrasekar** [comment]
Thanks for refactoring @besher-massri! @annapendleton can you take a look to make sure this doesn't affect the existing benchmarks?

**Review by annapendleton — CHANGES_REQUESTED** [review]
This is awesome work! Everything lgtm and shouldn't break existing functionality since it's just migrating locations. Only thing is we want to accurately represent what the infra does so people understand that it supports both GPU and TPU infra

File: benchmarks/gpu-benchmark.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+kubectl get nodes
+```
+
+### 2. Configure the cluster
+
+To configure the cluster to run inference workloads we need to set up workload identity, GCS Fuse and DCGM for GPU metrics. In the ai-on-gke/benchmarks folder run:
+
+```sh
+# Stage 2 configures the cluster for running inference workloads.
+cd infra/gpu-cluster/stage-2
```
[the remark this answers is not in the document]
Comment (besher-massri) [reply benchmarks/gpu-benchmark.md]: I went with using accelerator keyword instead, i.e. `accelerator-cluster` as opposed to the other `cpu-cluster`

File: benchmarks/README.md [hunk 3]
```diff
@@ -1,230 +1,10 @@
...
-# with future terraform commands.
-cd ../../infra/stage-2
-terraform destroy
-
-# Run destroy on infra/stage-1 resources
-cd ../stage-1
-terraform destroy
-```
+There are currently the following types of benchmarks available:
+- **GPU-based benchmark**: running inference servers on GPU based clusters. Can be found in `gpu-benchmark.md`
```
[the remark this answers is not in the document]
Comment (besher-massri) [reply benchmarks/README.md]: I pushed another commit, changing to accelerator and highlighting that both GPU and TPU clusters are supported. Changed some naming, minor rewording as well

**Comment by besher-massri** [comment]
Thanks for the review and comments! I pushed another commit, changing from gpu to accelerator and highlighting that both GPU and TPU clusters are supported. Changed some naming, minor rewording as well. Let me know if it's okay.

**Review by annapendleton — APPROVED** [review]

File: benchmarks/gpu-benchmark.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (annapendleton) [reply benchmarks/gpu-benchmark.md]: Thanks, that lgtm!

**Comment by annapendleton** [comment]
That looks great, thanks! Please make sure to resolve anything lingering with Andrew prior to submission :)

**Review by andrewsykim — APPROVED** [review]
LGTM, deferring merge to @genlu2011

File: 65k-benchmarks/simulated-ai-workload/README.md (hunk 1 above)
Comment (besher-massri) [reply 65k-benchmarks/simulated-ai-workload/README.md]: Is it okay to resolve this ?

**Pull request merged**